src/oc/web/actions/activity.cljs

```diff
@@ -239,14 +243,37 @@
       (dis/dispatch! [:modal-editing-activate]))
     (dis/dispatch! [:modal-editing-deactivate])))
 
+
+(declare entry-save)
+
 (defn entry-save-on-exit
```

Comment:
A `docstring` on this function and also some inline comments when you set/check/remove the initial-revision atom and/or the local cache would be great @thepug